packages/backend/routes/v1/crm/utils/hubspot.ts

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios';
+import qs from 'qs';
+import config from '../../../../config';
+import { logInfo, logError } from '../../../../helpers/logger';
+import { Prisma, xprisma } from '../../../../prisma/client';
+import { TP_ID } from '@prisma/client';
+import pubsub, { IntegrationStatusSseMessage, PUBSUB_CHANNELS } from '../../../../redis/client/pubsub';
+import { IntegrationAuthProps, mapIntegrationIdToIntegrationName } from '../../../../constants/common';
+import sendIntegrationStatusError from '../../sendIntegrationstatusError';
+
+const handleHubspotAuth = async ({
```

Comment:
@gold-olar This is a good to have but I think even the logic inside each of these handlers can be extracted out into a function? That way that function gets used in each of the handlers

Reply:
Hmmm... 
I think so, but not everything can be abstracted.
The basic logic in each of these handlers is pretty much based on each platform that we would be adding. 
The steps are pretty much to
- Make a network request with the code
- Fetch the user details with the access code
- Create the connection in revert db and return a success response


I feel like only the third step here can be further abstracted (`createRevertConnection`), and there would be no need to abstract any part of the first two as they are basically just axios calls and can really change depending on the platform (e.g see the Trello flow).

please lmk if this is how you were thinking of it as well.

Reply:
Yes @gold-olar I think step 3 can be abstracted away into its own util.

Reply:
Done ✅

While working on It, I thought of abstracting that call to `svix` as well, but I saw that the connection data that svix receives really carries across the different types of integrations. it would make it difficult to type it. that's why I left it out. 

the abstraction only logs error, and sends API responses as required.

Reply:
thanks @gold-olar! Let me dig deeper into this and get back to you. 🙌🏼